tests/regressions/regressions_determinise_state_limit_leak.c

```diff
@@ -0,0 +1,69 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include <assert.h>
+
+#include <re/re.h>
+#include <fsm/fsm.h>
+#include <fsm/bool.h>
+
+static const char *strings[] = {
```

Comment:
The input here is arbitrary, but one easy way to make a FSM whose state count increases significantly during determinisation is to combine several unanchored patterns.